### DISCUSSION TITLE
Camera button to capture photo or video not showing in Scene

### DISCUSSION THREAD

**Discussion by Gaurav-chopra** [google/model-viewer#1923] [Q&A]
I am using Oneplus 7t phone. When I use the glitch server, there the camera button to capture photo or video appears. But, when i upload these files on my own server. The camera button doesn't show. You can find the link as below:

https://portmethere.com/ar-saree

http://gaurav-ar.glitch.me/

Above are the two with same code, but different output of scene in same device.

**Comment by elalish** [comment]
Hmm, those two sites don't seem to have the same code from what I can see. The camera button in AR means you opened scene-viewer; the alternative on Android is WebXR. WebXR mode is the default and will fall back to scene-viewer if it fails, the most likely reason being http instead of https. Ensure you're always using `https`.

**Comment by Gaurav-chopra** [comment]
I have used the HTTPS in my mobile. It launches the Scene Viewer and augments the 3d model in real world. Just the camera button does'nt show up. It shows up when i use the glitch server even in http for glitch. You can check the above two url's to check.

**Comment by elalish** [comment]
That's not what I'm seeing. Your first link is https and launches webXR (no camera button), while your second link is http and launches scene-viewer (with camera button). If I change that second one to https, I get webXR, just like the first one. If you want to restrict your modes, just use the `ar-modes` attribute as documented. 

Out of curiosity, is the camera button an important feature to you? And how does it compare to the HTML customization of webXR? I ask because we're trying to prioritize different features to add.

**Comment by Gaurav-chopra** [comment]
Solved Thanks! I have removed the webXR mode from the code and it worked!. Yes the camera button is an important feature as it allows the user to capture photo or video and share it to others for feedback. For e.g. in ecommerce, they might want  to share it to someone with augmented model before purchase.

**Comment by Gaurav-chopra** [comment]
And is there any component for background audio in scene?

**Comment by elalish** [comment]
In 3D mode you can easily enough add audio as for any website. Scene-viewer does not support that kind of HTML customization, but webXR does.

**Comment by Gaurav-chopra** [comment]
The audio worked in Scene Viewer mode. The problem was i had not used any ar-mode so i think it was taking webXR mode as default and thus not playing sound. When i changed the priority to scene viewer, it worked! Thanks for your help! 
Is there any documentation for adding 3d models to location based targeting? Current documentation displays text.

**Comment by elalish** [comment]
> Is there any documentation for adding 3d models to location based targeting? Current documentation displays text.

I'm not sure what you mean. Can you be more specific?

**Comment by Gaurav-chopra** [comment]
I am talking about Location Tracking.
 `<a-text
        value="This content will always face you."
        look-at="[gps-camera]"
        scale="50 50 50"
        gps-entity-place="latitude: 21.1396932; longitude: 72.766791;"
      ></a-text>`

Can we have a 3dmodel instead of text content.

**Comment by elalish** [comment]
Oh, is this an A-frame thing? That sounds great, but you should probably ask on their github; it's unrelated to model-viewer, though they are both built on three.js.

**Comment by Gaurav-chopra** [comment]
Yeah right! Thanks for your help! Do we plan to have location based and image based tracking AR in model-viewer as well?

**Comment by elalish** [comment]
I'm not sure how location-based tracking would work in model-viewer's API; if you have an idea, please sketch it out for us. We are considering image-based tracking, but that'll have to land in webXR first.

**Comment by viniith21** [comment]
by using webxr it is possible to get only capture button in ar??

**Reply by elalish** [reply]
No, our WebXR mode doesn't have that feature yet - you'll need to use `ar-mode="scene-viewer"`.
